client/src/components/Login.vue

```diff
@@ -12,11 +12,11 @@
               </v-toolbar>
               <v-card-text>
                 <v-form>
-                  <v-text-field label="GAME ID" type="text" v-model="game_id"></v-text-field>
-                  <v-text-field label="USERNAME" type="text" v-model="user_name"></v-text-field>
-                  <v-btn @click.native="login" color="primary" :to="to">Join Game</v-btn>
-                  <v-btn color="primary" @click.native="newGame">Create new game</v-btn>
-                  <v-card v-if="status != ''">{{ status }}</v-card>
+                  <v-text-field id="gameId" label="GAME ID" type="text" v-model="game_id"></v-text-field>
```

Comment:
So far as I know, best practice is to not include `id`'s as it becomes very difficult to ensure uniqueness over time.

Reply:
Aaron, I think this is a good call out. I fully recognize this is counter to what most Computer Science courses teach. For my two cents on the matter. In my experience, I fully will admit that for development purposes, I would not recommend using `ids` for styling or identifying sections of code.  

However, from a testing methodology, not including `Id's` tends to require automation engineers relying on things like `css selectors`, `classes`, or even worse `xpath`.  From that perspective, while id's do run the risk of collision, in my opnion, they are by far the more reliable process in comparison to other options.

I have done/seen approaches of creating a pre-selected arbitrary html attribute for testing suites to rely on for selecting items. I agree on that as a better long term approach, and would be happy to investigate it if you feel it's preferable for Cypress. 

In short though, I genuinely am open if you have other more preferable solution.

Reply:
You can do some pretty neat things with xpath! `//button[contains(text(),"play")]` is a good example of what you can't do without it (so far as I know). Reference: https://devhints.io/xpath

That said, I totally agree that having `id`'s makes writing unit tests _so_ much easier. I guess, especially for unit tests on things like Vue components, having them makes sense and its easier to ensure uniqueness on a single component. 

This is an area where I, too, am hoping others have more experience and can offer guidance.

Reply:
ha ha ha for sure :) xpath is a pretty powerful tool. My experience has been that I'll only use it in a worst case scenario. (like what you described above). You can do some pretty stellar things with it that are practically impossible without it.

That being said. the more I thought about this. I ended up changing the `id's` to be an arbitrary HTML attribute `test-id`. and changed all of the queries to get elements to use a css selector to pull in those specific elements. I feel like this sets a better precedent for future testing, as it no longer completely ties it to the development process. (or randomly named id's)

I ended up prefixing component name in the `test-id` in order to specify which component it belongs to. 

Obviously these are all manual choices/paradigms. I equally am open to anyone else who may have a better way this can be handled.